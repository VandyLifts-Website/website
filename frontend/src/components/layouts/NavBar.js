/* Copyright P. Opiyo @2022 - All rights reserved */
import React from "react";
import { Link } from "react-router-dom";
import Navbar from "react-bootstrap/Navbar";
import Container from "react-bootstrap/Container";
import Form from "react-bootstrap/Form";

function NavBar(props) {
  return (
    <Navbar bg="light" variant="light">
      <Container>
        <Navbar.Brand href="/">
          <h2 style={{ color: "#cfae70" }}>VandyLifts</h2>
        </Navbar.Brand>
        <Form className="d-flex">
          {props.isLogin && (
            <Link to="/login" className="btn btn-outline-success">
              Sign In
            </Link>
          )}
          {props.isRegister && (
            <Link to="/register" className="ms-2 btn btn-outline-primary">
              Register
            </Link>
          )}
        </Form>
      </Container>
    </Navbar>
  );
}

export default NavBar;
