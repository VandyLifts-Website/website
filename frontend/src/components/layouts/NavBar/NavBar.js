/* Copyright P. Opiyo @2022 - All rights reserved */
import React from "react";
import { Link } from "react-router-dom";
import Navbar from "react-bootstrap/Navbar";
import Nav from "react-bootstrap/Nav";
import Container from "react-bootstrap/Container";
import Form from "react-bootstrap/Form";
import Button from "react-bootstrap/Button";

function NavBar(props) {
  return (
    <Navbar bg="light" variant="light">
      <Container>
        <Navbar.Brand href="/">
          <h2 style={{ color: "#cfae70" }}>VandyLifts</h2>
        </Navbar.Brand>
        {!props.isLoggedIn ? (
          <Form className="d-flex">
            <Link to="/signin" className="btn btn-outline-primary">
              Sign In
            </Link>
          </Form>
        ) : (
          <>
            <Nav
              className="me-auto"
              style={{ fontWeight: "bold", color: "purple" }}
            >
              <Nav.Link href="/profile">Profile</Nav.Link>
              <Nav.Link href="/organizations">Join an Organization</Nav.Link>
              <Nav.Link href="/about">Club Information</Nav.Link>
            </Nav>
            <Form className="d-flex">
              <Button href="/signin" variant="outline-warning">
                Sign Out
              </Button>
            </Form>
          </>
        )}
      </Container>
    </Navbar>
  );
}

export default NavBar;
