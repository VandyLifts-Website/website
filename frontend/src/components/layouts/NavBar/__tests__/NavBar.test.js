import {
  render,
  screen,
  cleanup,
  waitForElementToBeRemoved,
} from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import userEvent from "@testing-library/user-event";
import { MemoryRouter as Router } from "react-router-dom";
import NavBar from "../NavBar";

afterEach(() => {
  cleanup();
});

test("Renders sign in button when not logged in", () => {
  render(
    <Router>
      <NavBar isLoggedIn={false} />
    </Router>
  );
  // Check if the banner is rendered
  expect(screen.getByText(/VandyLifts/i)).toBeInTheDocument();
  const signinBtn = screen.getByRole("link", { name: "Sign In" });
  expect(signinBtn).toBeInTheDocument();
});

test("Logged out state matches snapshot", () => {
  const tree = renderer
    .create(
      <Router>
        <NavBar isLoggedIn={false} />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// test("Directs user to sign in when not logged in", async () => {
//   render(
//     <Router>
//       <NavBar isLoggedIn={false} />
//     </Router>
//   );

//   const signinBtn = screen.getByRole("link", { name: "Sign In" });
//   await waitForElementToBeRemoved
// })

test("Renders navigation links when logged in", () => {
  render(
    <Router>
      <NavBar isLoggedIn={true} />
    </Router>
  );
  // Check if the banner is rendered
  expect(screen.getByText(/VandyLifts/i)).toBeInTheDocument();
  expect(
    screen.getByRole("button", { name: "Fill a Survey" })
  ).toBeInTheDocument();
  expect(screen.getByRole("link", { name: "Profile" })).toBeInTheDocument();
  expect(
    screen.getByRole("link", { name: "Club Information" })
  ).toBeInTheDocument();
  expect(screen.getByRole("button", { name: "Sign Out" })).toBeInTheDocument();
});
