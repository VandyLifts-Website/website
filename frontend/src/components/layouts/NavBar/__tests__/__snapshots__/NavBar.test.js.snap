// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logged out state matches snapshot 1`] = `
<nav
  className="navbar navbar-expand navbar-light bg-light"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      <h2
        style={
          Object {
            "color": "#cfae70",
          }
        }
      >
        VandyLifts
      </h2>
    </a>
    <form
      className="d-flex"
    >
      <a
        className="btn btn-outline-primary"
        href="/signin"
        onClick={[Function]}
      >
        Sign In
      </a>
    </form>
  </div>
</nav>
`;
